src/controller/organization.controller.ts

```diff
@@ -0,0 +1,110 @@
+import { Controller, Get, Post, Put, Delete, Body, Param, UseGuards, Req, Res, HttpException, HttpStatus } from '@nestjs/common';
+import { OrganizationService } from 'src/organizations/organization.service';
+import { CreateOrganizationDto } from 'src/dto/create-organization.dto';
+import { UpdateOrganizationDto } from 'src/dto/update-organization.dto';
+import { JwtAuthGuard } from 'src/auth/jwt-auth.guard';
+import { check } from 'prettier';
+import { checkUser, hasScope } from 'src/utils/common.utils';
+import { Organization } from 'src/schema/organization.schema';
+import { HttpErrorByCode } from '@nestjs/common/utils/http-error-by-code.util';
+
+
+@Controller('organization')
+export class OrganizationController {
+    constructor(
+        private readonly organizationService: OrganizationService
+    ) { }
+
+    @Get()
+    @UseGuards(JwtAuthGuard)
+    async findAll(@Req() req, @Res() res) {
+        let where = {};
+        if (req.query.find) where = JSON.parse(req.query.find);
+
+        if (!hasScope(req.user, 'admin')) {
+            where['roles.members'] = req.user.sub;
+        }
+
+        const limit = req.query.limit || 0;
+        const skip = req.query.skip || 0;
+        const select =
+            req.query.select ||
+            'owner name created modified roles removed';
+        return res.json(
+            await this.organizationService.findAll(
+                where,
+                select,
+                skip,
+                null,
+                limit,
+            ));
+    }
+
+    @UseGuards(JwtAuthGuard)
+    @Get(':id')
+    async findOne(@Req() req, @Param('id') id: string) {
+        const organization: Organization = await this.organizationService.findOnebyId(id);
+
+        if (hasScope(req.user, "admin")) return organization;
+
+        const isMember = organization.roles.some(role => role.members.includes(req.user.sub));
```

Comment:
can this be encapsulated in the organization class?
```
.roles.some(role => role.members.includes(req.user.sub))
```
and other checks that you do for membership in an organization.